Arena/ArenaHooks.cs

```diff
@@ -192,6 +197,46 @@ private float SetRippleLevel(Func<Player, float> orig, Player self)
             return orig(self);
         }
 
+        private string? On_Options_LoadArenaSetup(On.Options.orig_LoadArenaSetup orig, Options self, string fallBack)
+        {
+            if (self.optionsLoaded && self.optionsFile != null && isArenaMode(out _))
+            {
+                return self.optionsFile.Contains("ArenaOnlineMeadowSetup") ? self.optionsFile.Get("ArenaOnlineMeadowSetup", "") : null;
+            }
+            return orig(self, fallBack);
+        }
+        private void On_Options_SaveArenaSetup(On.Options.orig_SaveArenaSetup orig, Options self, string arenaSetupStrings)
+        {
+            if (isArenaMode(out _))
+            {
+                if (self.optionsLoaded && self.optionsFile != null)
+                {
+                    self.optionsFile.Set("ArenaOnlineMeadowSetup", arenaSetupStrings, self.optionsFileCanSave ? Kittehface.Framework20.UserData.WriteMode.Immediate : Kittehface.Framework20.UserData.WriteMode.Deferred);
+                }
+                return;
+            }
+            orig(self, arenaSetupStrings);
+        }
+        private void IL_ArenaSetup_SaveToFile(ILContext il)
+        {
+            try
+            {
+                ILCursor cursor = new(il);
+                cursor.GotoNext(x => x.MatchCallvirt<Options>("SaveArenaSetup"));
+                cursor.GotoPrev(x => x.MatchLdarg(0));
+                cursor.Emit(OpCodes.Ldarg_0);
+                cursor.Emit(OpCodes.Ldloca, 0);
+                cursor.EmitDelegate(delegate (ArenaSetup self, ref string text)
+                {
+                    if (self is not ArenaOnlineSetup onlineSetup) return;
+                    text = onlineSetup.SetSaveStringFilter(text);
+                });
+            }
+            catch (Exception ex)
+            {
+                Error(ex);
+            }
+

```

Comment:
```suggestion
        }
```
missing a bracket. Probably got messed up during merging